app/src/main/java/com/amaze/filemanager/ui/fragments/MainFragment.java

```diff
@@ -1826,4 +1845,37 @@ public int getRootDrawable() {
   private boolean getBoolean(String key) {
     return getMainActivity().getBoolean(key);
   }
+
+  @Override
+  public void onGlobalLayout() {
+    if (mainFragmentViewModel.getColumns() == 0 || mainFragmentViewModel.getColumns() == -1) {
+      int screen_width = listView.getWidth();
+      int dptopx = Utils.dpToPx(getContext(), 115);
+      mainFragmentViewModel.setColumns(screen_width / dptopx);
+      if (mainFragmentViewModel.getColumns() == 0 || mainFragmentViewModel.getColumns() == -1) {
+        mainFragmentViewModel.setColumns(3);
+      }
+      if (!mainFragmentViewModel.isList()) {
+        mLayoutManagerGrid.setSpanCount(mainFragmentViewModel.getColumns());
+      }
+    }
+    if (!mainFragmentViewModel.isList()) {
+      loadViews();
+    }
+    if (android.os.Build.VERSION.SDK_INT >= JELLY_BEAN) {
+      mToolbarContainer.getViewTreeObserver().removeOnGlobalLayoutListener(this);
+    } else {
+      mToolbarContainer.getViewTreeObserver().removeGlobalOnLayoutListener(this);
+    }
+  }
+
+  public @Nullable MainFragmentViewModel getMainFragmentViewModel() {
+    try {
+      return new ViewModelProvider(this).get(MainFragmentViewModel.class);
+    } catch (Exception e) {
+      Log.e(getClass().getSimpleName(), "Failed to create viewmodel", e);
```

Comment:
Why the generic Exception catching?

Reply:
Please use a `Obejcts.requireNonNull()`

Reply:
Because it may throw fragment not initialised or any other fragment lifecycle exceptions in `new ViewModelProvider(this)`. Using `requireNonNull` won't help.

Reply:
The documentation doesn't say that [ViewModelProvider.get](https://developer.android.com/reference/androidx/lifecycle/ViewModelProvider#get(java.lang.Class%3CT%3E)) throws exceptions, did I miss something?

Reply:
It was throwing when i tested. I wasn't sure which ones that's why it's catching generic exception. 
Maybe the ones being thrown are runtime exception.

Reply:
> It was throwing when i tested. 

Detached fragment? This seems like another issue that's being shadowed here by this generic try catch.

Reply:
This look way to defensive, isn't there a better way to do this?

Reply:
The documentation asserts that the get() is nonnull, but here you are losing that guarantee, and then you have quite a lot of code rechecking to obtain again the guarantee of nonnullness.

Reply:
For detached fragment, without try/catch, I don't see any other fix apart from making major changes in codebase and use callbacks / observers.